src/main/java/com/webdev/cosmo/cosmobackend/service/user/UserService.java

```diff
@@ -0,0 +1,51 @@
+package com.webdev.cosmo.cosmobackend.service.user;
+
+import com.webdev.cosmo.cosmobackend.service.api.User;
+import com.webdev.cosmo.cosmobackend.service.common.mapper.UserMapper;
+import com.webdev.cosmo.cosmobackend.service.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.openapitools.model.UserModel;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    public UserModel save(User user) {
+        User savedUser = userRepository.save(user);
+
+        log.info("Saved User: " + savedUser);
+
+        return userMapper.mapToModel(savedUser);
+    }
+
+    public UserModel findByEmail(String email) {
+        User user = userRepository.findByEmail(email);
+        if(user != null)
+            return userMapper.mapToModel(user);
+        else throw new RuntimeException("Could not find user");
```

Comment:
Use exceptions which will be handled by us in @ControllerAdvice. RuntimExcpetion for such possible occurence is not recommended, as it would most likely end up with 500.